extensions/wikia/PhalanxII/PhalanxStatsSpecialController.class.php

```diff
@@ -12,45 +12,124 @@ function __construct( ) {
 		$this->wg->Out->addBacklinkSubtitle( $this->phalanxTitle );
 
 		if ( !$this->userCanExecute( $this->wg->User ) ) {
-			wfProfileOut( __METHOD__ );
 			$this->displayRestrictionError();
-			return;
+			return false;
 		}
 
 		$par = $this->getPar();
+
+		// Look for paths with a wiki ID like /wiki/Special:PhalanxStats/wiki/125951
 		if ( strpos( $par, 'wiki' ) === 0 ) {
 			// show per-wiki stats
-			list ( , $wikiId ) = explode( "/", $par, 2 );
-			$this->blockWikia( $wikiId );
-		} else {
-			$blockId = $this->wg->Request->getInt( 'blockId', intval( $par ) );
-			if ( !empty( $blockId ) ) {
-				// show block stats
-				$this->blockStats( $blockId );
-			} else {
-				// show help page
-				$this->forward( 'PhalanxStatsSpecial', 'help' );
-			}
+			return $this->forward( 'PhalanxStatsSpecial', 'blockWiki' );
+		}
+
+		// Look for paths with a block ID like /wiki/Special:PhalanxStats/123456
+		$blockId = $this->wg->Request->getInt( 'blockId', intval( $par ) );
+		if ( empty( $blockId ) ) {
+			// show help page
+			return $this->forward( 'PhalanxStatsSpecial', 'help' );
 		}
 
-		wfProfileOut( __METHOD__ );
+		// show block stats
+		return $this->blockStats( $blockId );
+	}
+
+	/**
+	 * Show a list of blocks for a specific wiki
+	 *
+	 * @return bool
+	 */
+	public function blockWiki() {
+
+		if ( !$this->userCanExecute( $this->wg->User ) ) {
+			$this->displayRestrictionError();
+			return false;
+		}
+	
+		$par = $this->getPar();
+		list ( , $wikiId ) = explode( '/', $par, 2 );
+
+		$wikiData = $this->getWikiData( $wikiId );
+		if ( empty( $wikiData ) ) {
+			return false;
+		}
+
+		// We have a valid id, change title to use it
+		$this->wg->Out->setPageTitle(
+			wfMessage( 'phalanx-stats-title' )->text() . ': ' . $wikiData['url']
+		);
+		$this->wg->Out->addBacklinkSubtitle( $this->title );
+
+		$this->getResponse()->setValues( [
+			'wikiData' => $wikiData,
+			'statsPager' => $this->buildWikiPager( $wikiId ),
+		] );
+
+		return true;
 	}
 
+	/**
+	 * Show information for a specific block ID
+	 *
+	 * @param $blockId
+	 * @return bool
+	 */
 	private function blockStats( $blockId ) {
 		$out = $this->getOutput();
 		$out->setPageTitle( $this->msg( 'phalanx-stats-title' )->text() . "#$blockId" );
 		$out->addBacklinkSubtitle( $this->title );
 
-		$data = Phalanx::newFromId( $blockId );
-
-		if ( !isset( $data["id"] ) ) {
+		$data = $this->getPhalanxData( $blockId );
+		if ( empty( $data ) ) {
 			$out->addWikiMsg( 'phalanx-stats-block-notfound', $blockId );
-			return false; // skip rendering
+			return false;
+		}
+
+		// pull these out of the array, so they don't get used in the top rows
+		$row = $data->toArray();
+		unset( $row['text'], $row['reason'], $row['comment'], $row['ip_hex'] );
+
+		$this->response->setValues( [
+			'firstRow' => $row,
+			'text' => $data['text'],
+			'reason' => $this->getParsedContent( $data, 'reason' ),
+			'comment' => $this->getParsedContent( $data, 'comment' ),
```

Comment:
Can use `Linker::formatComment` instead? It's a way lighter version of wikitext parser. I don't think that we need to handle templates or images here - pinging @Kirkburn to be sure :smile: